crates/provider/src/provider/prov_call.rs

```diff
@@ -33,12 +33,12 @@ where
     Conn: Transport + Clone,
     Params: RpcParam,
     Resp: RpcReturn,
-    Map: Fn(Resp) -> Output,
+    Map: Fn(Resp) -> Output + Clone,
 {
     /// An underlying call to an RPC server.
     RpcCall(RpcCall<Conn, Params, Resp, Output, Map>),
-    /// Call With Block
-    RpcCallWithBlock(RpcWithBlockFut<Conn, Params, Resp, Output, Map>),
+    /// RpcWithBlock
+    RpcWithBlock(RpcWithBlock<Conn, Params, Resp, Output, Map>),
```

Comment:
this looks like the same approach, just slightly modified.

all of this is a bit weird, because imo we need the inverse here, because rpcwithblock is just a helper type that drives something to completion. otherwise this doesn't solve the problem described in the issue. because this is restricted to rpcall:

https://github.com/alloy-rs/alloy/pull/788#pullrequestreview-2153603587